packages/suite/src/views/wallet/send/components/Options/components/BitcoinOptions/components/CoinControl.tsx

```diff
@@ -0,0 +1,218 @@
+}
+
+export const CoinControl = ({ close }: Props) => {
+    const [currentPage, setSelectedPage] = useState(1);
+
+    const { addToast, fetchTransactions } = useActions({
+        addToast: notificationActions.addToast,
+        fetchTransactions: fetchTransactionsThunk,
+    });
+
+    const { translationString } = useTranslation();
+
+    const {
+        account,
+        allUtxosSelected,
+        composedInputs,
+        dustUtxos,
+        errors,
+        getDefaultValue,
+        isCoinControlEnabled,
+        network,
+        outputs,
+        selectedUtxos,
+        spendableUtxos,
+        toggleCheckAllUtxos,
+        toggleCoinControl,
+    } = useSendFormContext();
+
+    const { areSatsUsed } = useBitcoinAmountUnit(account.symbol);
+
+    const inputs = isCoinControlEnabled ? selectedUtxos : composedInputs;
+
+    const getTotal = (amounts: number[]) =>
+        amounts.reduce((previous, current) => previous + current, 0);
+    const getFormattedAmount = (amount: number) =>
+        formatNetworkAmount(amount.toString(), account.symbol);
+
+    // calculate and format amounts
+    const totalInputs = getTotal(inputs.map(input => Number(input.amount)));
+    const totalOutputs = getTotal(
+        outputs.map((_, i) => Number(getDefaultValue(`outputs[${i}].amount`, ''))),
+    );
+    const totalOutputsInSats = areSatsUsed ? totalOutputs : totalOutputs * 10 ** network.decimals;
+    const missingToInput = totalOutputsInSats - totalInputs;
+    const formattedTotal = getFormattedAmount(totalInputs);
+    const formattedMissing = getFormattedAmount(missingToInput);
+    const isMissingToAmount = missingToInput > 0;
+    const isMissingVisible =
+        isCoinControlEnabled &&
+        (isMissingToAmount ||
+            !!errors.outputs?.some(
+                error =>
+                    ((error?.amount as TypedFieldError)?.message as ExtendedMessageDescriptor)
+                        ?.id === 'TR_NOT_ENOUGH_SELECTED',
+            ));
+    const missingToInputId = isMissingToAmount ? 'TR_MISSING_TO_INPUT' : 'TR_MISSING_TO_FEE';
+
+    // pagination
+    const totalItems = account.utxo?.length || 0;
+    const utxosPerPage = SETTINGS.TXS_PER_PAGE;
+    const showPagination = totalItems > utxosPerPage;
+    const lastSpendableUtxoIndex = currentPage * utxosPerPage;
+    const spendableUtxosOnPage = spendableUtxos.slice(
+        lastSpendableUtxoIndex - utxosPerPage,
+        lastSpendableUtxoIndex,
+    );
+    const lastDustUtxoIndex = currentPage * utxosPerPage - spendableUtxos.length;
+    const dustOnPage = dustUtxos.slice(
+        lastDustUtxoIndex - utxosPerPage,
+        lastDustUtxoIndex > 0 ? lastDustUtxoIndex : 0,
+    );
+
+    useEffect(() => {
+        try {
+            fetchTransactions({
+                account,
+                page: 2,
+                perPage: SETTINGS.TXS_PER_PAGE,
+                noLoading: true,
+                recursive: true,
+            });
+        } catch {
```

Comment:
This catch will never catch, because **thunks from RTK are always resolved**. Please read this https://redux-toolkit.js.org/api/createAsyncThunk#unwrapping-result-actions 

`unwrap` should solve issue here